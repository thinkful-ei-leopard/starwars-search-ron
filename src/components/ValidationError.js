import React from 'react';

export default function ValidationError(props) {
    // If message is a string, display the message, otherwise if it is undefined return an empty fragment.
    if(props.message) {
        return (
            <div className="validation_error">{props.message}</div>
        );
    }

    return <></>;
}