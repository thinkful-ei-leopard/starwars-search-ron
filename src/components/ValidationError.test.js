
import React from 'react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import ValidationError from './ValidationError';

describe('<ValidationError />', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<ValidationError />, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    it('renders this UI as expected', () => {
        const tree = renderer.create(<ValidationError />).toJSON();
        expect(tree).toMatchSnapshot();
    });    
});