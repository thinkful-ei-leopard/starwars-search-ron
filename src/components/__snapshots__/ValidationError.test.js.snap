// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ValidationError /> renders this UI as expected 1`] = `null`;
